### ISSUE TITLE
Cannot start session, headers already sent.

### ISSUE THREAD

**Issue by andrewlimaza** [strangerstudios/paid-memberships-pro#1172]
**Describe the bug**
Warning that headers are already sent for `pmpro_start_session`.

**To Reproduce**

(Test on PHP 7.0+)

1. Login to your site with a membership account.
2. Have debugging enabled, and go to checkout page for a different level.
3. Warning should be logged.

Reference to code:  https://github.com/strangerstudios/paid-memberships-pro/blob/dev/includes/sessions.php#L21

Tested loading on `init` instead and resolves the issue.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only Paid Memberships Pro plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Memberlite](https://wordpress.org/themes/memberlite/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by mircobabini** [comment]
It also happens when you run the WP Crons from the command line (crontab) instead of internal scheduler, as reported [here](https://github.com/strangerstudios/paid-memberships-pro/issues/232).

In general, `session_status() == PHP_SESSION_NONE` is not enough.
- sometimes may need `headers_sent( $filename, $linenum )`
- sometimes may need `defined('STDIN')` or `php_sapi_name() == 'cli'`

[Check this up](https://www.php.net/manual/en/function.session-status.php#113468). A bit old and needs tests but it should work.

By the way, I see that the suggested way now is: `if (!session_id()) session_start()`

**Comment by mircobabini** [comment]
Mh, could also be a matter of timing. This morning I found this in logs:

```
[03-May-2020 09:15:01 UTC] PHP Warning:  session_start(): Cannot start session when headers already sent in ...../wp-content/plugins/paid-memberships-pro/includes/sessions.php on line 30
[03-May-2020 09:15:57 UTC] PHP Warning:  Headers not sent but sent an error: Array
(
    [type] => 2
    [message] => unlink(...../wp-content/wflogs/template.0877404001588497357.tmp): No such file or directory
    [file] => ...../wp-content/plugins/wordfence/lib/wordfenceClass.php
    [line] => 2138
)
```

Of course I added a line or two to dump these message about who sent the headers.
What if we run that script about session initializing jut a bit earlier?

**Comment by CarlPotak** [comment]
I'm getting this warning every two to three minutes, 24/7. That's up to 30 requests an hour. I have also isolated that this occurs from PMPro. 

This is my error:
PHP Warning session_start(): Cannot start session when headers already sent in /home/startu59/public_html/wp-content/plugins/paid-memberships-pro/includes/sessions.php on line 21

I did some research to find this other possible explanation, but that's it:
[https://www.rodude.com/warning-cannot-start-session-when-headers-already-sent/]

Any headway on this? I'd love to see a fix for this sooner than later.
